sparkplug-launcher/src/main/scala/springnz/sparkplug/client/ClientExecutor.scala

```diff
@@ -0,0 +1,87 @@
+package springnz.sparkplug.client
+
+import akka.actor.TypedActor.PreStart
+import akka.actor._
+import com.typesafe.config.{ Config, ConfigFactory }
+import springnz.sparkplug.executor.MessageTypes._
+import springnz.util.{ Pimpers, Logging }
+
+import scala.concurrent._
+import scala.concurrent.duration._
+import scala.util.Try
+
+trait ClientExecutor {
+  def execute[A](pluginClass: String, data: Option[Any]): Future[A]
+  def shutDown(): Unit
+}
+
+object ClientExecutor extends Logging {
+  import Constants._
+  import Coordinator._
+  import Pimpers._
+
+  def create(config: Config = ConfigFactory.load().getConfig(defaultConfigSectionName)) = {
+
+    case class ClientActors(parent: ActorRef, coordinator: ActorRef)
+
+    val actorSystem: ActorSystem = {
+      Try { ActorSystem(actorSystemName, config) }
+        .withErrorLog("Unable to create ActorSystem")
+        .get
+    }
+
+    implicit val executionContext = actorSystem.dispatcher
+
+    // Create a parent actor to receive messages (so they don't get sent to deadLetters)
+    // This actor then creates the Coordinator
+    val actors: Future[ClientActors] = {
+      val readyPromise = Promise[ActorRef]()
+
+      val parentActor = actorSystem.actorOf(Props(new Actor with Logging with PreStart {
```

Comment:
This parentActor seems redundant. Errors that occur in the Coordinator could just be logged in the Coordinator. If the Coordinator sends a message to an actor that is offline, this parentActor won't receive the dead letter since it is not subscribed to the dead letter event stream. Instead of the parentActor, we could create a DeadLetterSubscriber actor as a sibling of the Coordinator.